bluecellulab/analysis/inject_sequence.py

```diff
@@ -113,21 +116,90 @@ def run_stimulus(
     simulation.run(stimulus.stimulus_time, cvode=cvode)
 
     # Retrieve simulation results
+    recordings = []
     current = np.array(current_vector.to_python())
-    voltage = cell.get_voltage_recording(cell.sections[recording_section], recording_segment)
-    time = cell.get_time()
+    for recording_section, recording_segment in recording_locations:
+        recording_location = f"{recording_section}({str(recording_segment)})"
+        voltage = cell.get_voltage_recording(cell.sections[recording_section], recording_segment)
+        time = cell.get_time()
 
-    if len(time) != len(voltage) or len(time) != len(current):
-        raise ValueError("Time, current, and voltage arrays are not the same length")
+        if len(time) != len(voltage) or len(time) != len(current):
+            raise ValueError("Time, current, and voltage arrays are not the same length")
 
-    if enable_spike_detection:
-        results = cell.get_recorded_spikes(location=recording_location, threshold=threshold_spike_detection)
-        if results is not None:
-            spikes = np.array(results)
-        else:
-            spikes = None
+        if enable_spike_detection:
+            results = cell.get_recorded_spikes(location=recording_location, threshold=threshold_spike_detection)
+            if results is not None:
+                spikes = np.array(results)
+            else:
+                spikes = None
+
+        recordings.append(
+            Recording(current=current, voltage=voltage, time=time, spike=spikes)
+        )
+
+    return recordings
+
+
+def run_stimulus(
+    template_params: TemplateParams,
+    stimulus: Stimulus,
+    section: str,
+    segment: float,
+    cvode: bool = True,
+    add_hypamp: bool = True,
+    recording_section: str = "soma[0]",
+    recording_segment: float = 0.5,
+    enable_spike_detection: bool = False,
+    threshold_spike_detection: float = -30.0,
```

Comment:
Do we have a threshold voltage hardcoded somewhere else (threshold current calculation?) in bluecellulab? If yes, then we should have one global variable, e.g. `DEFAULT_SPIKE_DETECTION_VOLTAGE`. I see -20 mV in `bluecellulab/tools/` in `detect_spike()`. In efel, the threshold voltage is -20 mV. Maybe keep -20 mV here. I think thalamus models might need the lower voltage(?).

Reply:
I don't believe we have any threshold voltage hardcoded in BCL. I changed the default value to -20